src/material-experimental/mdc-button/button.scss

```diff
@@ -45,3 +45,15 @@
   right: -$mdc-button-outlined-border-width;
   border: none;
 }
+
+// MDC expects button icons to contain this HTML content:
+// ```html
+//   <span class="mdc-button__icon material-icons">favorite</span>
+// ```
+// However, Angular Material expects a `mat-icon` instead. The following
+// will extend the `mdc-button__icon` styling to the mat icon. Note that
+// the extended styles inherently only match icons that nest themselves in
+// a parent `mdc-button`.
+.mat-icon {
+  @extend .mdc-button__icon;
```

Comment:
I think that we've been trying to avoid using `@extend` on concrete selectors, because it can end up bringing in unnecessary styles. Should we leave a TODO somewhere to expose a mixin through MDC?